### DISCUSSION TITLE
[Question] should I be able to use direnv ‘system' ?

### DISCUSSION THREAD

**Discussion by andrewcrook** [asdf-community/asdf-direnv#87] [Q&A] [answered]
I setup my asdf global python to ‘system’ so the default uses python2 from macOS and python3 from homebrew.

I already have direvn installed via homebrew so I tried the same by setting it to ‘system'.
However, asdf cannot find direnv 

`asdf exec direnv`
`unknown command: direnv. Perhaps you have to reshim?`

reshim makes no difference.

Therefore, I am wondering if it is because the plugin asdf-direnv hasn’t accounted for this configuration?

Update: I have tried https://github.com/asdf-community/asdf-direnv/discussions/58#discussioncomment-205115

`mkdir -p ~/.asdf/installs/direnv/system/env`

then reshiming but I get the same

**Comment by RCristiano** [comment]
It seems asdf is not managing direnv.
What's in your `~/.tool-versions`?

If there is no line with direnv version use: `asdf global direnv system`

Hope this helps

**Comment by michi-zuri** [comment, marked as answer]
May I suggest you try hooking up direnv with 
```eval "$(direnv hook bash)"``` instead of 
```eval "$(asdf exec direnv hook bash)"```
Since you are using the system-installed direnv, there is no need for asking asdf to tell you direnv's path, just use call direnv directly from your system's path. 

For direnv to understand the `use asdf` instruction you'd still need to add the asdf hook to the direnv config:
```
# File: ~/.config/direnv/direnvrc
source "$(asdf direnv hook asdf)"  
```

I couldn't test my suggestions, since my OS (Solus) doesn't offer a direnv package. Please let us know if it didn't work. @RCristiano's suggestion may or might not be required, depending if the asdf shims are before or after your system paths. I personally make sure that asdf shims are the last thing on my path to avoid for example anaconda shims of anything and everything to shadow my system bins.

**Reply by RCristiano** [reply]
How do you ensure that asdf shims are the last thing out of the system's path?

**Reply by michi-zuri** [reply]
instead of sourcing `. ~/.asdf/asdf.sh` I just set the path myself. For system-installed direnv this could look like this:
```
### asdf setup
# do NOT source the asdf.sh script, instead add the shims path manually (and even that is optional)
# . ~/.asdf/asdf.sh
PATH=~/.asdf/bin:$PATH:~/.asdf/shims 
# enable auto completions for asdf
. ~/.asdf/completions/asdf.bash
### direnv setup for direnv managed by the system (no need for direnv system in .tool-versions)
eval "$(direnv hook bash)"
# you don't want the direnv alias function in this case:
# direnv() { asdf exec direnv "$@"; }
```

Ps. 
In the direnv config file you still need asdf in front of direnv because this is the script provided by the asdf direnv plugin that needs to be found. 

PsPs. 
If you ever need to use anaconda or miniconda with direnv, the init script for conda should go into ~/.config/direnv/direnvrc instead of the ~/.bashrc to make the activate commands work in an environment file (like `conda activate` in `.envrc`).

**Reply by RCristiano** [reply]
Thank you very much, it will help me a lot!
I use "fish" and I will build on your tips, this will help me to use direnv with poetry which is still giving me problems working with direnv.
